### DISCUSSION TITLE
Memory issue using Windows (yes I know it is not supported)

### DISCUSSION THREAD

**Discussion by eziolanza** [MIC-DKFZ/nnUNet#1704] [Q&A] [answered]
Hello everyone,
I'm using 32GB ram RTX 3090 Intel I7 CPU. I was able to train all 2d folds on a 180 cases dataset. Was previously able to train 3d_lowres on a small 30 cases dataset; however I am not able to train the 180 dataset using 3d_lowres , I get this error:

"OSserror winerror 8 not enough memory resources are available to process this command " 

No matter how I increase the memory swap file this remains. Any workaround for this? Increase RAM? Any way to tweak the system so that It will work with the available hardware?

Thanks for your help

**Comment by ancestor-mithril** [comment]
You can set the environment variable `nnUNet_n_proc_DA`, which controls the number of Data Augmentation workers, to a lower value. The default is 8, I think that 4 is ok. 
You can also set the environment variable `nnUNet_def_n_proc` (default: 8) to a lower value such as 4.

**Reply by eziolanza** [reply]
Already tried with no luck. I think that controls the amount of CPU cores claimed, this seems rather a RAM related issue ?

**Reply by ancestor-mithril** [reply]
The number of processes used influence the RAM. If it loads 8 samples it occupies more RAM than loading 4 samples. 
Try setting `nnUNet_n_proc_DA` to 0. And please check that nnUNet sees the correct value of `nnUNet_n_proc_DA`, because on Windows setting an evironment variable is not as straightforward. 
You can do so by using a small python script that prints `os.environ["nnUNet_n_proc_DA"]`.

**Reply by eziolanza** [reply]
I tried with 0, checked  Get-ChildItem env: and was correctly updated. Still states memory available is insufficient with OSErorr: WinError 8. 
I also tried maxing up the swap memory, without luck.
Other ideas?

**Reply by ancestor-mithril** [reply]
Normally, I would expect nnUNet to work with 32 GB, that's why use `python -c "import os; print(os.environ['nnUNet_n_proc_DA'])"` to check if it is seen by the python intepreter.

If you use `nnUNet_keep_files_open=1`, remove it and use `nnUNet_keep_files_open=0`. 
 That's all I think of.

**Reply by eziolanza** [reply]
I'll try that. Curiously enough, the issue arises only when I train 3d_lowres. both 2d and 3d_fullres (unexpectedly) work fine. 
Maybe a bug?

**Reply by ancestor-mithril** [reply, marked as answer]
It could be. If 3d_fullres works, I recommend using it as it's usually better than using 3d_lowres.
